core/src/tile/mapTile.cpp

```diff
@@ -47,13 +44,15 @@ void MapTile::draw(const Style& _style, const View& _view) {
 
         std::shared_ptr<ShaderProgram> shader = _style.getShaderProgram();
 
-        glm::dmat4 modelViewProjMatrix = _view.getViewProjectionMatrix() * m_modelMatrix;
+        // Apply tile-view translation to the model matrix
+        const auto& viewOrigin = _view.getPosition();
+        m_modelMatrix[3][0] = m_tileOrigin.x - viewOrigin.x;
```

Comment:
Yup, sounds good. I know we did discuss doing something like this when we found that the tesselator only handles floats. :D